### PR TITLE
Add support for llvm.tan.*

### PR DESCRIPTION
**Pull request by maarquitos14**

### CODE REVIEW

File: test/llvm-intrinsics/tan.ll [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+define dso_local spir_func float @foo(float %x) local_unnamed_addr #0 {
+entry:
+  %0 = call float @llvm.tan.f32(float %x)
+  ret float %0
+}
+
+; Function Attrs: nounwind readnone speculatable willreturn
+declare float @llvm.tan.f32(float) #1
+
+attributes #0 = { nounwind readnone "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "unsafe-fp-math"="false" "use-soft-float"="false" }
```
Comment (MrSidims) [inline test/llvm-intrinsics/tan.ll:21]: nit: no need for attributes and metadata in this test.

**Review by MrSidims — COMMENTED** [review]
Please consider adding the intrinsic to getNativeBuiltinIdForIntrinsic following logic of this patch:
commit 6417adab353c6863d40365fe2b9c37e98f440f71
Author: Victor Mustya <victor.mustya@intel.com>
Date:   Thu Nov 24 04:12:12 2022 -0800

    Translate LLVM intrinsics into native_* OpenCL instructions (#1729)

File: test/llvm-intrinsics/tan.ll [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir"
+
+; CHECK: ExtInst {{[0-9]+}} {{[0-9]+}} {{[0-9]+}} tan 
+
+; Function Attrs: nounwind readnone
+define dso_local spir_func float @foo(float %x) local_unnamed_addr #0 {
+entry:
+  %0 = call float @llvm.tan.f32(float %x)
```
Comment (MrSidims) [inline test/llvm-intrinsics/tan.ll:14]: nit: that is up to you, I see that we are here a bit inconsistent and some tests have only scalar fp32 versions others have: fp16,fp32,fp64 versions + 1 vector of fp32 version. Up to you to decide if extra testing is needed

Comment (maarquitos14) [reply test/llvm-intrinsics/tan.ll]: I have extended the test to cover more cases.

File: test/llvm-intrinsics/tan.ll (hunk 1 above)
Comment (maarquitos14) [reply test/llvm-intrinsics/tan.ll]: Removed.

**Comment by maarquitos14** [comment]
> Please consider adding the intrinsic to getNativeBuiltinIdForIntrinsic following logic of this patch: commit [6417ada](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/6417adab353c6863d40365fe2b9c37e98f440f71) Author: Victor Mustya [victor.mustya@intel.com](mailto:victor.mustya@intel.com) Date: Thu Nov 24 04:12:12 2022 -0800
> 
> ```
> Translate LLVM intrinsics into native_* OpenCL instructions (#1729)
> ```

Done.

File: test/llvm-intrinsics/tan.ll [hunk 3]
```diff
@@ -0,0 +1,77 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: spirv-val %t.spv
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir"
+
+; CHECK: ExtInstImport [[ExtInstSetId:[0-9]+]] "OpenCL.std"
```
Comment (MrSidims) [inline test/llvm-intrinsics/tan.ll:9]: I'm fine with this notation, but just FYI for numerical variables for LIT regexp the following is considered a bit better: [[#ExtInstSetId:]] and use of the var will be [[#ExtInstSetId]]

**Review by MrSidims — APPROVED** [review]

**Comment by MrSidims** [comment]
I'll leave a day before merge in case if anybody would like to review

File: test/llvm-intrinsics/tan.ll (hunk 3 above)
Comment (vmaksimo) [reply test/llvm-intrinsics/tan.ll:9]: +1 To use numerical variables through all the test.

**Review by vmaksimo — COMMENTED** [review]
LGTM

File: test/llvm-intrinsics/tan.ll (hunk 3 above)
Comment (maarquitos14) [reply test/llvm-intrinsics/tan.ll]: Done, thanks both for the suggestion.

**Review by svenvh — APPROVED** [review]

**Pull request merged**